lib/geomancer/geo_json/feature_set.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Geomancer.GeoJson.FeatureSet do
+  @moduledoc false
+  alias Geomancer.GeoJson.Feature
+
+  @type t() :: [Feature.t()]
+
+  @spec reduce([struct()], keyword()) :: [t()]
+  def reduce(geometry, opts \\ []) do
+    type = Keyword.get(opts, :type, "Point")
+    keys = Keyword.get(opts, :properties, [])
+
+    {features, _, _} = Enum.reduce(geometry, {[], type, keys}, &feature_reducer/2)
+    Enum.reverse(features)
+  end
+
+  defp feature_reducer({%Exshape.Shp.Header{}, _}, acc), do: acc
+
+  defp feature_reducer({shape, values}, {features, type, keys}) do
+    props = parse_properties(keys, values)
+    coords = parse_coordinates(shape)
+    bbox = parse_bounding_box(shape)
+
+    {[Feature.new(type, bbox, props, coords) | features], type, keys}
+  end
+
+  defp parse_properties(keys, values) do
+    values
+    |> Enum.map(&trim_dbf_value/1)
+    |> Enum.zip(keys)
+    |> Enum.map(fn {val, key} -> {key, val} end)
```

Comment:
This made me do a double take. But on second pass it 'just makes sense'.

Reply:
Thanks for pointing this one out. I've got an idea to clean it up.